src/client/java/net/neoforged/neoforge/client/model/quad/MutableQuad.java

```diff
@@ -0,0 +1,459 @@
+    public void setPosComponent(int vertexIndex, int componentIndex, float value) {
+        positions[vertexIndex].setComponent(componentIndex, value);
+    }
+
+    /**
+     * Sets a vertex's position.
+     */
+    public void setPos(int vertexIndex, float x, float y, float z) {
+        positions[vertexIndex].set(x, y, z);
+    }
+
+    /**
+     * Sets a vertex's position.
+     */
+    public void setPos(int vertexIndex, Vector3fc position) {
+        positions[vertexIndex].set(position);
+    }
+
+    /**
+     * {@return the x-component of a vertex's normal or NaN if the normal is undefined}
+     */
+    public float getNormalX(int vertexIndex) {
+        return getNormalComponent(vertexIndex, 0);
+    }
+
+    /**
+     * {@return the y-component of a vertex's normal or NaN if the normal is undefined}
+     */
+    public float getNormalY(int vertexIndex) {
+        return getNormalComponent(vertexIndex, 1);
+    }
+
+    /**
+     * {@return the z-component of a vertex's normal or NaN if the normal is undefined}
+     */
+    public float getNormalZ(int vertexIndex) {
+        return getNormalComponent(vertexIndex, 2);
+    }
+
+    /**
+     * {@return a component of a vertex's normal or NaN if the normal is undefined}
+     *
+     * <p>See <a href="#components">components</a> for the mapping of index to component.
+     */
+    public float getNormalComponent(int vertexIndex, int componentIndex) {
+        var packedNormal = normals[vertexIndex];
+        if (BakedNormals.isUnspecified(packedNormal)) {
+            return Float.NaN;
+        } else {
+            return BakedNormals.unpackComponent(packedNormal, componentIndex);
+        }
+    }
+
+    /**
+     * {@return a mutable copy of a vertex's normal}
+     * <p>
+     * Pass a non-null destination vector to avoid allocation of a new vector.
+     */
+    public Vector3f copyNormal(int vertexIndex, @Nullable Vector3f dest) {
+        if (dest == null) {
+            dest = new Vector3f();
+        }
+        return BakedNormals.unpack(normals[vertexIndex], dest);
+    }
+
+    public Vector3f copyNormal(int vertexIndex) {
+        return copyNormal(vertexIndex, new Vector3f());
+    }
+
+    public void setFrom(BakedQuad quad) {
+        for (int i = 0; i < 4; i++) {
+            positions[i] = new Vector3f(quad.position(i));
+            normals[i] = quad.bakedNormals().normal(i);
+            colors[i] = quad.bakedColors().color(i);
+        }
+        tintIndex = quad.tintIndex();
+        direction = quad.direction();
+        sprite = quad.sprite();
+        shade = quad.shade();
+        lightEmission = quad.lightEmission();
+        hasAmbientOcclusion = quad.hasAmbientOcclusion();
+    }
+
+    /**
+     * Sets the positions of this quad to form a rectangle on the given block side using a coordinate-system matching
+     * the default orientation of sprites in Vanilla block-models.
+     * <p>
+     * Inspired by the Fabric Rendering API method {@code square}.
```

Comment:
```suggestion
     * Inspired by the Fabric Renderer API method {@code square}.
```
(nitpick, I know :D)